common/src/main/java/com/wynntils/overlays/BombBellOverlay.java

```diff
@@ -75,22 +86,37 @@ public BombBellOverlay() {
     @Override
     public void render(
             GuiGraphics guiGraphics, MultiBufferSource bufferSource, DeltaTracker deltaTracker, Window window) {
+        List<TextRenderTask> renderTasks = groupBombs.get()
+                ? Models.Bomb.getBombBells().stream()
+                        .filter(bombInfo -> {
+                            BombType bombType = bombInfo.bomb();
+                            Supplier<Boolean> bombTypeSupplier = bombTypeMap.get(bombType);
+                            return bombTypeSupplier != null && bombTypeSupplier.get();
+                        })
+                        .collect(Collectors.groupingBy(BombInfo::bomb))
+                        .values()
+                        .stream()
+                        .flatMap(list -> list.stream().sorted(comparator).limit(maxBombs.get()))
+                        .map(bombInfo -> new TextRenderTask(bombInfo.asString(), textRenderSetting))
+                        .toList()
+                : Models.Bomb.getBombBells().stream()
+                        .filter(bombInfo -> {
+                            BombType bombType = bombInfo.bomb();
+                            Supplier<Boolean> bombTypeSupplier = bombTypeMap.get(bombType);
+                            return bombTypeSupplier != null && bombTypeSupplier.get();
+                        })
+                        .sorted(comparator)
+                        .limit(maxBombs.get())
+                        .map(bombInfo -> new TextRenderTask(bombInfo.asString(), textRenderSetting))
+                        .toList();
```

Comment:
At least the first filter and the last map can be extracted into common code.

This also brings the question whether this should be actually updated on render. I'd move this to update in tick.